src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -2204,6 +2443,381 @@ def construct_command(self, task_exec, task_args, task_numcores,
+            config_yarn_site()
+
+            logger.info('Start Formatting DFS')
+            namenode_format = os.system(hadoop_home + '/bin/hdfs namenode -format -force')
+            logger.info('DFS Formatted. Starting DFS.')
+            hadoop_start = os.system(hadoop_home + '/sbin/start-dfs.sh')
+            logger.info('Starting YARN')
+            yarn_start = os.system(hadoop_home + '/sbin/start-yarn.sh')
+
+            #-------------------------------------------------------------------
+            # Creating user's HDFS home folder
+            logger.debug('Running: %s/bin/hdfs dfs -mkdir /user'%hadoop_home)
+            os.system('%s/bin/hdfs dfs -mkdir /user'%hadoop_home)
+            uname = commands.getstatusoutput('whoami')
+            logger.debug('Running: %s/bin/hdfs dfs -mkdir /user/%s'%(hadoop_home,uname[1]))
+            os.system('%s/bin/hdfs dfs -mkdir /user/%s'%(hadoop_home,uname[1]))
+            check = commands.getstatusoutput('%s/bin/hdfs dfs -ls /user'%hadoop_home)
+            logger.info(check[1])
+            # FIXME YARN: why was the scheduler configure called here?  Configure
+            #             is already called during scheduler instantiation
+            # self._scheduler._configure()
+
+            service_url = node_name + ':54170'
+            rm_url      = node_name
+            launch_command = yarn_home + '/bin/yarn'
+            rm_ip = node_name
+
+          
+        # The LRMS instance is only available here -- everything which is later
+        # needed by the scheduler or launch method is stored in an 'lm_info'
+        # dict.  That lm_info dict will be attached to the scheduler's lrms_info
+        # dict, and will be passed around as part of the opaque_slots structure,
+        # so it is available on all LM create_command calls.
+        lm_info = {'service_url'   : service_url,
+                   'rm_url'        : rm_url,
+                   'hadoop_home'   : hadoop_home,
+                   'rm_ip'         : rm_ip,
+                   'name'          : lrms.name,
+                   'launch_command': launch_command,
+                   'nodename'      : lrms.node_list[0] }
+
+        return lm_info
+
+
+    # --------------------------------------------------------------------------
+    #
+    @classmethod
+    def lrms_shutdown_hook(cls, lm_info, logger):
+        if 'name' not in lm_info:
+            raise RuntimeError('rm_ip not in lm_info for %s' \
+                    % (self.name))
+
+        if lm_info['name'] != 'YARNLRMS':
+            logger.info('Stoping YARN')
+            os.system(lm_info['hadoop_home'] + '/sbin/stop-yarn.sh')
+
+            logger.info('Stoping DFS.')
+            os.system(lm_info['hadoop_home'] + '/sbin/stop-dfs.sh')
+
+            logger.info("Deleting HADOOP files from temp")
+            os.system('rm -rf /tmp/hadoop*')
+            os.system('rm -rf /tmp/Jetty*')
+            os.system('rm -rf /tmp/hsperf*')
```

Comment:
How does  that work for multiple users, in those cases where the agent runs on a head node?  They can't all end up in the same dir structure, can they?